Hamilton_Matthew/history.py

```diff
@@ -0,0 +1,173 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        #request.session stuff
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.        
+                    
+        if not request.user.is_authenticated:
+            request.session['new_prod'] = -1
+
+        new_prod_id = int(request.session['new_prod'])
+        print ("new_prod_id: ")
+        print(new_prod_id)
+        
+        if new_prod_id > 0:
+            print("woah")
```

Comment:
Very clever to set new_prod_id to 0 in index.py so this code is only run if detail.py passes a product id. 
Remove the print statements in this file. 
This would be a great file to add some comments explaining what things are doing.

Reply:
All good points